doc/howtos/software-update-server.rst

```diff
@@ -0,0 +1,281 @@
+.. _software-update-server:
+
+Software Update: Building an Ostro |trade| OS  Repository
+#########################################################
+
+This technical note provides instructions for creating
+a repository used for updating devices based on Ostro OS.
+
+Prerequisites
+=============
+
+- additional (fast) disk space: the SW Update tool is provided as
+  an additional layer, therefore it doesn't require any additional
+  element.
+  The process of generating the bundles and updates can be very
+  disk intensive and dramatic speed improvements can come from
+  using a fast disk storage system, including use of SSD and RAID
+  arrays.
+  An execution round will also produce various rootfs directories
+  as intermediate artefacts.
+  The actual disk footprint depends on the bundles selected and
+  defined and will be larger than that produced by a simple typical
+  distribution build based on OECore or Yocto Project.
+- signing keys (alternatively testing keys from the :file:`swupd-server`
+  project can be used).
+
+Bundles definition
+==================
+
+The documentation for using SW Update in Yocto Project can
+be found in the `swupd layer`_ and is only summarized here.
+These instructions assume that the work will happen on top of an
+image flavor that has swupd enabled, for example by building an image,
+such as with::
+
+   $ bitbake ostro-image-swupd
+
+or by downloading a pre-compiled image with ``-swupd`` in its name
+from the `Ostro OS download server`_
+
+.. _`Ostro OS download server`: https://download.ostroproject.org
+
+These are the main steps:
+
+- In ``conf/local.conf`` or equivalent location, set the
+  `OS_VERSION` variable to an integer value.
+  This number must be increased before each build that generates
+  swupd update artefacts.
```

Comment:
This is not required. The automatic default in local builds works well enough to make explicit setting of OS_VERSION an optional step.